### ISSUE TITLE
Modify "Find It" feature to treat search terms as AND, not OR.

### ISSUE THREAD

**Issue by eric-gilbertson** [RocketMan/zookeeper#173]
Currently find it returns items that contain any of the words in the user entered phrase, eg searching for "Take off my dress" returns all items that contain the word dress instead of the Ana Egge song of that name. The utility of this feature would be greatly increased if it returned only items that contained all the user entered words.

**Comment by RocketMan** [comment]
You do know that if you put the phrase in quotes exactly as you typed above, it will return only hits with that exact phrase, right?

As for OR vs AND, sometimes people do not know exactly what they are looking for and throw out a bunch of things to the best of their recollection.  For example, a search phrase might include 'red dress' but it was actually 'crimson dress'.  AND insures the later will not match, and requires the person to retool their search terms.  More work for the user.

People may be using Find It! in ways you don't anticipate.  The feature has been around for a while and there are expectations around how it works.  Before you make any changes, please raise this on the music list so people who use the feature can weigh in.

**Comment by RocketMan** [comment]
Actually, looking at the code just now, it is already doing an AND, not an OR.

As proof, type White.  See the results.  Then White Horse.  Notice you have fewer, not more results.

In your example, 'take off my dress', everything is a stopword except for 'dress'.  If you put that phrase in double quotes, it will find just your playlists with 'take off my dress', but if you just type it as-is, you get a list of things which match 'dress'.

(I'm not sure why 'take' is being treated as a stopword; I am going to look into that further.)

So bottom line, Find It! already does AND not OR.

**Comment by RocketMan** [comment]
> (I'm not sure why 'take' is being treated as a stopword; I am going to look into that further.)

'Take' is, in fact, a stopword:
https://dev.mysql.com/doc/refman/8.0/en/fulltext-stopwords.html#fulltext-stopwords-stopwords-for-myisam-search-indexes

**Comment by RocketMan** [comment]
Given that we already do AND and not OR, shall we close this?

Or do you want to keep it open and revisit the stop lists?  I am gobsmacked how many words are in there, that in our case, probably should not be.  Zk is not a huge database, so I don't think indexing things like 'take' or 'besides' or any of a number of words of that sort will cause us undue strain.

**Comment by RocketMan** [comment]
I have a prepared a significantly reduced stopword list and am tracking in #176 

To minimize any possible disruption, I will defer restarting mysqld and rebuilding the indexes until tomorrow morning after midnight.

**Comment by RocketMan** [comment]
OK, after the ops changes of #176, you will find that your sample phrase, 'take off my dress', even WITHOUT the quotation marks now yields a much more reasonable result:

![afbeelding](https://user-images.githubusercontent.com/6860356/89025340-8e69cf00-d31e-11ea-8414-b8c6445aee5b.png)

**NOTE**: Now, the only stopword in the phrase is 'my', so you do get some extra things like 'take off your dress'.  (Previously, *everything* was a stopword except for dress).
